demos/common/combined/aws_iot_combined_demo.c

```diff
@@ -0,0 +1,991 @@
+    io_conf.pin_bit_mask = (1<<GPIO_INPUT_IO_0);
+    //set as input mode
+    io_conf.mode = GPIO_MODE_INPUT;
+    //enable pull-up mode
+    io_conf.pull_up_en = 1;
+    gpio_config(&io_conf);
+
+    AwsIotLogInfo( "vBLEButtonTask: start task\n" );
+
+    while( true )
+    {
+        if (buttonWasPushed())
+        {
+            if (AwsIotNetworkManager_GetEnabledNetworks() & AWSIOT_NETWORK_TYPE_BLE) 
+            {
+                AwsIotLogInfo( "vBLEButtonTask: BLE Button OFF\n");
+                AwsIotNetworkManager_DisableNetwork(AWSIOT_NETWORK_TYPE_BLE);
+            }
+            else 
+            {
+                /* TODO: Once BLE starts, WiFi can be reprovisioned which triggers a WiFi disconnect call back. */
+                AwsIotLogInfo( "vBLEButtonTask: BLE Button ON\n");
+                AwsIotNetworkManager_EnableNetwork(AWSIOT_NETWORK_TYPE_BLE);
+            }
+        }
+    }
+    AwsIotLogInfo( "vBLEButtonTask: stop task\n" );
+}
+
+void vProducerTask(void * pvParameters )
+{
+    uint16_t counter = 0;
+    TickType_t qsendDelay = pdMS_TO_TICKS( demoQUEUE_SEND );
+    /* Create the queue used to pass data to publish task. */
+    AwsIotLogInfo( "vProducerTask: start task\n" );
+    while( true )
+    {
+        if( xQueueSend( xTelemetryQueue, &counter, pdMS_TO_TICKS(1000) ) != pdPASS )
+        {
+            AwsIotLogError(( "vProducerTask: Failed to send data to TelemetryQueue\n" ) );
+        }
+        else 
+        {
+            ++counter;
+        }   
+        /*AwsIotLogInfo( "vProducerTask: produced %d\n", (int) counter);*/
+        vTaskDelay( qsendDelay );
+    }
+    AwsIotLogInfo( "vProducerTask: stop task\n" );
+}
+
+AwsIotMqttError_t prxPublishMQTTMessage( const char* pcMesg, size_t xLength )
+{
+    AwsIotMqttPublishInfo_t xPublishInfo = AWS_IOT_MQTT_PUBLISH_INFO_INITIALIZER;
+    AwsIotMqttReference_t xOperationLock = AWS_IOT_MQTT_REFERENCE_INITIALIZER;
+    AwsIotMqttError_t xStatus;
+
+    xPublishInfo.QoS = 1;
+    xPublishInfo.pTopicName = demoMQTT_TOPIC;
+    xPublishInfo.topicNameLength = strlen( demoMQTT_TOPIC );
+
+    xPublishInfo.pPayload = ( void * ) pcMesg;
+    xPublishInfo.payloadLength = xLength;
+    xPublishInfo.retryLimit = 0;
+            /* WiFI is started before vStartCombinedDemo(); therefore, this Give is required. */
+    xPublishInfo.retryMs = 0;
+
+
+    if( demoMQTT_QOS == 0 )
+    {
+        xStatus = AwsIotMqtt_Publish(
+                xConnection.xMqttConnection,
+                &xPublishInfo,
+                0,
+                NULL,
+                NULL );
+    }
+    else if( demoMQTT_QOS == 1 )
+    {
+        /* PUBLISH a message with timeout */
+        xStatus = AwsIotMqtt_TimedPublish(
+            xConnection.xMqttConnection,
+            &xPublishInfo,
+            0,
+            demoPUBLISH_TIMEOUT_DELAY_MS );
+
+        AwsIotLogInfo("prxPublishMQTTMessage: pub string = %s", xPublishInfo.pPayload);
+    }
+    else if( demoMQTT_QOS == 2 )
```

Comment:
demoMQTT_QOS  == 2 can be a timed publish as well, so can be merged with previous if statement.